app/assets/javascripts/discourse/app/components/form-template-field/upload.js

```diff
@@ -0,0 +1,69 @@
+import Component from "@ember/component";
+import UppyUploadMixin from "discourse/mixins/uppy-upload";
+import { computed } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import { dasherize } from "@ember/string";
+import { isAudio, isImage, isVideo } from "discourse/lib/uploads";
+
+export default class FormTemplateFieldUpload extends Component.extend(
+  UppyUploadMixin
+) {
+  @tracked uploadValue;
+  @tracked fileUploadElementId = this.attributes?.label
+    ? `${dasherize(this.attributes.label)}-uploader`
+    : `${this.elementId}-uploader`;
+  @tracked fileInputSelector = `#${this.fileUploadElementId}`;
+  @tracked id = this.fileUploadElementId;
+  @tracked uploadComplete = false;
+  @tracked uploadedFiles = [];
+  @tracked disabled = this.uploading;
+
+  @computed("uploading", "uploadValue")
```

Comment:
I'm wondering why we need `@computed` here when `uploadValue` and `disabled` are both tracked properties.

Reply:
Yes, I'm not sure why it isn't auto-tracking when I leave it as a native getter... Perhaps because it's a classic component, that's just using native class syntax? (Needed because we implement a `UppyUploadMixin`).

Reply:
Oo icic. Thank you for clarifying. Not a big problem here I was just wondering why we still need `@computed`.